proxy/http2/Http2Stream.cc

```diff
@@ -327,7 +327,12 @@ Http2Stream::do_io_close(int /* flags */)
     if (parent && this->is_client_state_writeable()) {
       // Make sure any trailing end of stream frames are sent
       // Wee will be removed at send_data_frames or closing connection phase
-      static_cast<Http2ClientSession *>(parent)->connection_state.send_data_frames(this);
+      Http2SendDataFrameResult result = static_cast<Http2ClientSession *>(parent)->connection_state.send_data_frames(this);
+      if (result == Http2SendDataFrameResult::NO_WINDOW) {
+        Http2StreamDebug("cancel closing");
+        closed = false;
+        return;
```

Comment:
When will `do_io_close` be called again in this case? On receiving WINDOW_UPDATE?

Reply:
Yes, the window update should cause the data to be sent again.  If the end of the data has been transmitted into the stream before the do_io_close, eventually the end frame will be sent, and the stream shutdown will be started again.  If the do_io_close was called in response to an error case, I would be worried that we would have to wait for a timeout before the stream finishes.